### ISSUE TITLE
Add scaffolding for config

### ISSUE THREAD

**Issue by thaiat** [mcfly-io/generator-mcfly#21]

**Comment by vshanbha** [comment]
Perhaps not relevant to this issue but I have had to add support in my gulp tasks for env specific config. I couldn't find documentation on this generator around same so I assume it is not yet available out of the box. 
e.g. I may have different API endpoints for my Production env vs dev , test env.
This is not same as --mode property that is provided to the gulp tasks. 
Perhaps this stack overflow link does a better job at explaining what I mean. 
[Gulp build with browserify environment variable](http://stackoverflow.com/questions/28099041/gulp-build-with-browserify-environment-variable)

To get this done In my actual project I have something like this in the gulp_tasks/common/constants.js 
```
var argv = require('yargs').argv;
....
....
var environment = argv.env ? argv.env : 'dev';
    if (!(environment === 'prod' || environment === 'dev' || environment === 'test')) {
        environment = 'dev'; // Force env to be dev for invalid values
    }
....
....
          script: {
            dest: 'scripts',
            apiConfigSrc: './' + clientFolder + '/config/config_' + environment + '.js',
            apiConfigDest: './' + clientFolder + '/scripts/api/constants'
        }

```

Elsewhere in the gulp_tasks/tasks/dist.js I have 

```
    gulp.src(constants.script.apiConfigSrc)
        .pipe(rename('config.js'))
        .pipe(gulp.dest(constants.script.apiConfigDest));
```

This way I can have config_dev.js , config_prod.js, config_test.js in my source which overwrite the config.js file and the same gets used by rest of the code to locate env specific constants.

My suggestion is to have a way to do something similar out of the box.
If it is already present (and I have missed out the docs) I welcome comments directing me to the appropriate files. I also welcome any comments on whether above way is a good way (or not) to do env specific config.

**Comment by jskrzypek** [comment]
That's a very project-specific option, since (I think) you're talking about the API for the backend that you're using, right?

**Comment by vshanbha** [comment]
@jskrzypek 
Sorry if my communication is misleading.
I am not talking about API for backend. 
Surely most projects would have some configuration which varies according to the target deployment env (dev vs prod). For me it is the API endpoint location. For other projects it may be something else.
My config.js will have env specific properties that are associated with my project. Same way there will be other projects which will have their own env specific config.
My point is that the generator should generate some placeholder code so users can write whatever they want in their config.js.
Thus when when I to `gulp dist --mode=prod --env=<dev | test | prod>` it should generate properly uglified code injected with the env specific config. 
Hope that makes sense

**Comment by thaiat** [comment]
this is already supported.
Take a look at gulps/common/helper.js -> getEnvifyVars function

**Comment by vshanbha** [comment]
@thaiat  Thanks for the reference. I went through the method but I am a newbie to Angular and Node.js so I am not quite sure I fully understand how I can set this up for my purpose. Is there any documentation / wiki that lays it out in simpler terms ?

**Comment by thaiat** [comment]
you can check out ho envify works here : https://github.com/hughsk/envify
